src/main/java/org/sagebionetworks/web/client/SynapseJSNIUtilsImpl.java

```diff
@@ -444,5 +444,13 @@ public void replaceHistoryState(String token) {
 	private static native void _replaceHistoryState(String token)/*-{
 		$wnd.history.replaceState( {} , '', '#'+token );
 	}-*/;
-	
+
+	@Override
+	public void pushHistoryState(String token) {
+		_pushHistoryState(token);
+	}
+
+	private static native void _pushHistoryState(String token)/*-{
+		$wnd.history.pushState( {} , '', '#'+token );
+	}-*/;
 }
```

Comment:
In console, there is an error in this file (line 374), the 2nd line of this method:
public final static native void _consoleError(String message) /*-{
		console.error(message);
	}-*/;
Maybe this is why the pushState() did not work.